### PR TITLE
[docker] Make python version configurable on docker image

### PR DESCRIPTION
**Pull request by Leemoonsoo**
## Why are these changes needed?

To help build a docker image with different python versions.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: build-docker.sh [hunk 1]
```diff
@@ -37,8 +38,13 @@ case $key in
     echo "not implemented, just hardcode me :'("
     exit 1
     ;;
+    --python-version)
+    # Python version to install. e.g. 3.7.7
+    shift
+    PYTHON_VERSION=$1
+    ;;
```
Comment (ijrsvt) [inline build-docker.sh:45]: Can you just add a comment stating that changing python versions may require a different wheel

**Review by ijrsvt — APPROVED** [review]
Looks great! Thank you so much for adding this!!

File: build-docker.sh (hunk 1 above)
Comment (Leemoonsoo) [reply build-docker.sh]: Thank @ijrsvt for reviewing this.
Just addressed your comment.

Comment (ijrsvt) [reply build-docker.sh]: Perfect--looks good!

**Comment by simon-mo** [comment]
Wow this is awesome!

**Pull request merged**